### DISCUSSION TITLE
Are there any support for mongodb like storage?

### DISCUSSION THREAD

**Discussion by Raj2032** [gluesql/gluesql#536] [Q&A] [answered]
Hey guys I wanted to ask does gluesql supports nosql similar to mogodb?

If so then is this something that sled supports or I would have to use something else?

Which type of database would be the fastest way to read and write?

**Comment by panarch** [comment, marked as answer]
> Hey guys I wanted to ask does gluesql supports nosql similar to mogodb?

There is plans to support more storages including some nosql databases.
Not now, but it will become possible to have mongodb as a storage of GlueSQL.
In a next few months, there are plans to support primary key and schema-less data. (planning to support all of them until Sep, this year)
Then, supporting NoSQL backend for GlueSQL becomes possible.

If you want some unstructured data support for now, then you can consider to use `MAP` and `LIST` types.
Full schema-less (unstructrued) data support is not supported yet.

> Which type of database would be the fastest way to read and write?

This really depends on use cases.
Focusing on read, write or append-only, etc..
It's hard to simply say the answer what would be the best.

**Comment by KyGost** [comment]
Note that one of the (*many*) wonderful ways @panarch has written this library is that the extensibility of storages is massive.

One could set up an external crate, say `gluesql-mongodb`, which adds a mongodb store.

**Reply by KyGost** [reply]
Example:
https://github.com/gluesql/gluesql/tree/main/storages%2Fsled-storage

(Functions as a separate crate, is just included in the main crate)

**Reply by Raj2032** [reply]
Interesting mate thanks :)
